ludwig/utils/data_utils.py

```diff
@@ -197,6 +197,10 @@ def read_stata(data_fp, df_lib):
     return df_lib.read_stata(data_fp)
 
 
+def read_hdf5(data_fp, df_lib):
```

Comment:
df_lib is unused, maybe we can have kwargs instead? also please add type hints

Reply:
tried to keep consistent with the other `read` functions, which take in `(data_fp, df_lib)`, but can remove